fvm/evm/emulator/config.go

```diff
@@ -0,0 +1,176 @@
+package emulator
+
+import (
+	"math"
+	"math/big"
+
+	"github.com/ethereum/go-ethereum/common"
+	"github.com/ethereum/go-ethereum/core"
+	"github.com/ethereum/go-ethereum/core/vm"
+	"github.com/ethereum/go-ethereum/crypto"
+	"github.com/ethereum/go-ethereum/params"
+)
+
+var (
+	FlowEVMTestnetChainID = big.NewInt(666)
+	FlowEVMMainnetChainID = big.NewInt(777)
+)
+
+// Config sets the required parameters
+type Config struct {
+	// Chain Config
+	ChainConfig *params.ChainConfig
+	// EVM config
+	EVMConfig vm.Config
+	// block context
+	BlockContext *vm.BlockContext
+	// transaction context
+	TxContext *vm.TxContext
+	// base unit of gas for direct calls
+	DirectCallBaseGasUsage uint64
+}
+
+var zero = uint64(0)
+var DefaultChainConfig = &params.ChainConfig{
+	ChainID: FlowEVMTestnetChainID, // default is testnet
+
+	// Fork scheduling based on block heights
+	HomesteadBlock:      big.NewInt(0),
+	DAOForkBlock:        big.NewInt(0),
+	DAOForkSupport:      false,
+	EIP150Block:         big.NewInt(0),
+	EIP155Block:         big.NewInt(0),
+	EIP158Block:         big.NewInt(0),
+	ByzantiumBlock:      big.NewInt(0), // already on Byzantium
+	ConstantinopleBlock: big.NewInt(0), // already on Constantinople
+	PetersburgBlock:     big.NewInt(0), // already on Petersburg
+	IstanbulBlock:       big.NewInt(0), // already on Istanbul
+	BerlinBlock:         big.NewInt(0), // already on Berlin
+	LondonBlock:         big.NewInt(0), // already on London
+	MuirGlacierBlock:    big.NewInt(0), // already on MuirGlacier
+
+	// Fork scheduling based on timestamps
+	ShanghaiTime: &zero, // already on Shanghai
+	CancunTime:   &zero, // already on Cancun
+	PragueTime:   &zero, // already on Prague
+}
+
+func defaultConfig() *Config {
+	return &Config{
+		ChainConfig: DefaultChainConfig,
+		EVMConfig: vm.Config{
+			NoBaseFee: true,
+		},
+		TxContext: &vm.TxContext{},
+		BlockContext: &vm.BlockContext{
+			CanTransfer: core.CanTransfer,
+			Transfer:    core.Transfer,
+			GasLimit:    math.MaxUint64, // block gas limit
```

Comment:
hmm, I don't think we need to use it anywhere else, this is the default gas limit on the block level (we just set it to this to prevent block limit enforcement).

Reply:
updated.